Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,125 @@
+    /// with a constant volume for each bar.
+    /// </summary>
+    public class VolumeRenkoConsolidator : DataConsolidator<BaseData>
+    {
+        private VolumeRenkoBar _currentBar;
+        private decimal _barSize;
+
+        /// <summary>
+        /// Gets a clone of the data being currently consolidated
+        /// </summary>
+        public override IBaseData WorkingData => _currentBar;
+
+        /// <summary>
+        /// Gets <see cref="VolumeRenkoBar"/> which is the type emitted in the <see cref="IDataConsolidator.DataConsolidated"/> event.
+        /// </summary>
+        public override Type OutputType => typeof(VolumeRenkoBar);
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        public event EventHandler<VolumeRenkoBar> DataConsolidated;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VolumeRenkoConsolidator"/> class using the specified <paramref name="barSize"/>.
+        /// </summary>
+        /// <param name="barSize">The constant volume size of each bar</param>
+        public VolumeRenkoConsolidator(decimal barSize)
+        {
+            _barSize = barSize;
+        }
+
+        /// <summary>
+        /// Updates this consolidator with the specified data
+        /// </summary>
+        /// <param name="data">The new data for the consolidator</param>
+        public override void Update(BaseData data)
+        {
+            var close = data.Price;
+            var dataType = data.GetType();
+
+            decimal volume;
+            decimal open;
+            decimal high;
+            decimal low;
+
+            if (dataType == typeof(TradeBar))
+            {
+                volume = ((TradeBar)data).Volume;
+                open = ((TradeBar)data).Open;
+                high = ((TradeBar)data).High;
+                low = ((TradeBar)data).Low;
+            }
+            else if (dataType == typeof(Tick))
+            {
+                // Only include actual trade information
+                if (((Tick)data).TickType != TickType.Trade)
+                {
+                    return;
+                }
+
+                volume = ((Tick)data).Quantity;
+                open = close;
+                high = close;
+                low = close;
+            }
+            else
+            {
+                throw new ArgumentException("VolumeRenkoConsolidator() must be used with TradeBar or Tick data.");
+            }
+
+            if (_currentBar == null)
+            {
+                _currentBar = new VolumeRenkoBar(data.Symbol, data.Time, data.EndTime, _barSize, open, high, low, close, 0);
+            }
+            var volumeLeftOver = _currentBar.Update(data.EndTime, high, low, close, volume);
+            while (volumeLeftOver >= 0)
+            {
+                OnDataConsolidated(_currentBar);
+                _currentBar = _currentBar.Rollover();
```

Comment:
Don't think rollover is accurate always, it will reuse the same high/low of the consolidated bar but those could of be older of previous bars, it should use the current data to create the new bar

Reply:
yah, I will use `Close` for both attributes on rolling over